pkg/types/validation/installconfig.go

```diff
@@ -543,6 +543,18 @@ func validateImageContentSources(groups []types.ImageContentSource, fldPath *fie
 func validateNamedRepository(r string) error {
 	ref, err := dockerref.ParseNamed(r)
 	if err != nil {
+		// If a mirror name is provided without the named reference,
+		// then the name is not considered canonical and will cause
+		// an error. e.g. registry.lab.redhat.com:5000 will result
+		// in an error. Instead we will check whether the input is
+		// a valid hostname as a workaround.
+		if err == dockerref.ErrNameNotCanonical {
```

Comment:
Can we call `dockerref.ParseNormalizedNamed` instead of `dockerref.ParseNamed` to avoid the canonical check? Or is there an example of a non-canonical URI that would be accepted that we want to reject?

Reply:
After calling `ParseNormalizedNamed`, `ParseNamed` also checks to see if the original string is the same as what `ParseNormalizedNamed` spits out. If they're the same then the original input was kosher and all is well. If you don't pass anything that is not canonical, then `ParseNormalizedNamed` makes changes and we will encounter the `dockerref.ErrNameNotCanonical` error.

https://github.com/containers/image/blob/b7361977328b2d8cd0a75333e4124589405cd45f/docker/reference/reference.go#L246

e.g.
`host.example.com:5000/library/package` : Input is Canonical. All is well.
`foo/bar` transforms to `docker.io/library/foo/bar` : Name transformed into a canonical string but fails string check
host.example.com:5000 transforms to docker.io/library/host.example.com:5000: This fails the string check in ParseNamed and we get the canonical error. This is a corner case, we want to succeed. So we perform an additional URI check here to see if its a valid hostname and continue if true. Or else we exit.